tools/rocketchat/incoming.js

```diff
@@ -0,0 +1,345 @@
+      "**Pushed to " +
+      "[" +
+      request.content.repository.full_name +
+      "](" +
+      request.content.repository.url +
+      "):" +
+      request.content.ref.split("/").pop() +
+      "**\n\n";
+
+    for (var i = 0; i < commits.length; i++) {
+      var commit = commits[i];
+      var shortID = commit.id.substring(0, 7);
+      var a = "[" + shortID + "](" + commit.url + ") - " + commit.message;
+      if (commits.length > 1) {
+        output.push(a);
+      } else {
+        var output = a;
+      }
+    }
+
+    if (commits.length > 1) {
+      text += output.reverse().join("\n");
+    } else {
+      text += output;
+    }
+
+    return {
+      content: {
+        attachments: [
+          {
+            thumb_url: user.avatar_url,
+            text: text,
+            fields: []
+          }
+        ]
+      }
+    };
+  }, // End GitHub Push
+
+  /* NEW PULL REQUEST */
+  pull_request(request) {
+    const user = request.content.sender;
+    const attachment = {
+      author_icon:
+        "https://cloud.githubusercontent.com/assets/51996/13893698/c047133c-ed2e-11e5-9233-13622bcb9b7b.png",
+      author_name:
+        "#" +
+        request.content.pull_request.number +
+        " - " +
+        request.content.pull_request.title,
+      author_link: request.content.pull_request.html_url
+    };
+
+    // To disable some webhooks I've placed a _ at the beginning of the action.
+    if (
+      request.content.action == "opened" ||
+      request.content.action == "reopened" ||
+      request.content.action == "edited" ||
+      request.content.action == "_synchronize" ||
+      request.content.action == "_review_requested"
+    ) {
+      var body = request.content.pull_request.body;
+    } else if (request.content.action == "_labeled") {
+      var body =
+        "Current labels: " +
+        getLabelsField(request.content.pull_request.labels).value;
+    } else if (
+      request.content.action == "_assigned" ||
+      request.content.action == "_unassigned"
+    ) {
+      // Note that the issues API only gives you one assignee.
+      var body =
+        "Current assignee: " + request.content.pull_request.assignee.login;
+    } else if (request.content.action == "_closed") {
+      if (request.content.pull_request.merged) {
+        var body = "Merged by: " + request.content.pull_request.merged_by.login;
+      } else {
+        var body = "Closed.";
+      }
+    } else {
+      return {
+        error: {
+          success: false,
+          message: "Unsupported pull request action"
+        }
+      };
+    }
+
+    const action = request.content.action.capitalizeFirstLetter();
+
+    let text = "Pull request [" + action + "]\n\n";
+    text += "# " + request.content.pull_request.number;
+    text +=
+      " - **[" +
+      request.content.pull_request.title +
+      "](" +
+      request.content.pull_request.html_url +
+      ")**\n\n" +
+      body;
+
+    // switch (request.content.action) {
```

Comment:
Leaving this in for now as it has references to variables that would be useful when I have to to modify this script.